app/(api)/_datalib/collections/findItem.ts

```diff
@@ -0,0 +1,33 @@
+import { getClient } from '@utils/mongodb/mongoClient.mjs';
+import { NotFoundError } from '@utils/response/Errors';
+
+export async function findCollectionItems(collection: never, query = null) {
+  try {
+    const client = await getClient();
+    const db = client.db;
+    const reqCollection = await db.getCollection(collection);
+    const reqDocument = await reqCollection.find(query).toArray();
+    if (!reqDocument || reqDocument.length === 0) {
+      return {
+        ok: false,
+        body: null,
+        error: { code: 404, message: 'No items found in collection' },
+      };
+    }
+    return { ok: true, body: reqDocument, error: null };
+  } catch (error) {
+    if (error instanceof NotFoundError) {
+      return {
+        ok: false,
+        body: null,
+        error: { code: 404, message: error.message },
```

Comment:
I like how you return error codes. I feel like this could be implemented a little more modularly if you define custom errors like I did in `/(api)/_utils/response`. This is fine for now tho :)